config/module/tree.go

```diff
@@ -162,21 +173,38 @@ func (t *Tree) Load(storage getter.Storage, mode GetMode) error {
 	t.lock.Lock()
 	defer t.lock.Unlock()
 
-	// discover where our modules are going to be stored
-	s := newModuleStorage(storage)
+	s := newModuleStorage(storage, mode)
+
+	children, err := t.getChildren(s)
```

Comment:
It's a bit confusing here that `t.getChildren` and `t.children` both exist but refer to objects at a different stage of the lifecycle. I'm not sure what's better, though... maybe calling this `t.findChildren` helps slightly, since it makes it sound less like just an accessor method for `t.children`. :man_shrugging:

Reply:
Good point about these sounding like accessors.  I'll keep it in mind as I continue to clean up this file.